src/javascript/_common/auth.js

```diff
@@ -93,6 +94,21 @@ export const requestOauth2Logout = onWSLogoutAndRedirect => {
 
 export const requestSingleLogout = async (onWSLogoutAndRedirect) => {
     const requestSingleLogoutImpl = async () => {
+        // Check if TMB is enabled first
+        if (TMB.isTMBEnabled()) {
+            const clientAccounts = JSON.parse(localStorage.getItem('client.accounts') || '{}');
+            const isClientAccountsPopulated = Object.keys(clientAccounts).length > 0;
+            const isCallbackPage = window.location.pathname.includes('callback');
+            const isEndpointPage = window.location.pathname.includes('endpoint');
+
+            if (isClientAccountsPopulated && !isCallbackPage && !isEndpointPage) {
+                await TMB.handleTMBLogout();
+                await requestOauth2Logout(onWSLogoutAndRedirect);
```

Comment:
[R] TMB logout does not require oauth2 logout

```suggestion
                await onWSLogoutAndRedirect();
```